### PR TITLE
A Photo Tool (Libre) version 1.0.5-3

### PR DESCRIPTION
**Pull request by aphototool**
### Please confirm your submission meets all the criteria

- [X] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [X] My pull request follows the instructions at [App Submission][submission].
- [X] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [X] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [X] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [X] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [X] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

File: me.ahola.aphototoollibre.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+app-id: me.ahola.aphototoollibre
+runtime: org.kde.Platform
+runtime-version: '5.15-22.08'
+sdk: org.kde.Sdk
+command: aphototoollibre
+finish-args:
+  - --share=ipc
+  - --socket=x11
+  - --socket=wayland
+  - --env=XDG_SESSION_TYPE=x11
```
Comment (hfiguiere) [inline me.ahola.aphototoollibre.yml:10]: no. don't set this.

File: me.ahola.aphototoollibre.yml [hunk 2]
```diff
@@ -0,0 +1,27 @@
+app-id: me.ahola.aphototoollibre
+runtime: org.kde.Platform
+runtime-version: '5.15-22.08'
+sdk: org.kde.Sdk
+command: aphototoollibre
+finish-args:
+  - --share=ipc
+  - --socket=x11
+  - --socket=wayland
```
Comment (hfiguiere) [inline me.ahola.aphototoollibre.yml:9]: it's either `x11` alone, or `fallback-x11` with wayland

File: me.ahola.aphototoollibre.yml [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+  - --device=dri
+modules:
+  - name: aphototoollibre
+    buildsystem: simple
+    sources:
+      - type: archive
+        url: https://github.com/aphototool/A-Photo-Tool-Libre/archive/refs/tags/v1.0.5-3.tar.gz
+        sha256: f28201302d36c510e542e7d11653afddba0a2df4a829bc5e810f7ed5c365b5d3
+    build-commands:
+      - qmake
```
Comment (hfiguiere) [inline me.ahola.aphototoollibre.yml:21]: it should be `buildsystem: qmake`

File: me.ahola.aphototoollibre.yml [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+      - make
+      - install -Dm755 /run/build/aphototoollibre/bin/aphototoollibre /app/bin/aphototoollibre
+      - install -Dm644 /run/build/aphototoollibre/flatpak/desktop/me.ahola.aphototoollibre.desktop /app/share/applications/me.ahola.aphototoollibre.desktop
+      - install -Dm644 /run/build/aphototoollibre/flatpak/desktop/me.ahola.aphototoollibre.svg /app/share/icons/hicolor/scalable/apps/me.ahola.aphototoollibre.svg
+      - install -Dm644 /run/build/aphototoollibre/flatpak/desktop/me.ahola.aphototoollibre.appdata.xml /app/share/metainfo/me.ahola.aphototoollibre.appdata.xml
```
Comment (hfiguiere) [inline me.ahola.aphototoollibre.yml:26]:
```suggestion
      - install -Dm755 ./bin/aphototoollibre /app/bin/aphototoollibre
      - install -Dm644 ./flatpak/desktop/me.ahola.aphototoollibre.desktop /app/share/applications/me.ahola.aphototoollibre.desktop
      - install -Dm644 ./flatpak/desktop/me.ahola.aphototoollibre.svg /app/share/icons/hicolor/scalable/apps/me.ahola.aphototoollibre.svg
      - install -Dm644 ./flatpak/desktop/me.ahola.aphototoollibre.appdata.xml /app/share/metainfo/me.ahola.aphototoollibre.appdata.xml
```

**Comment by hfiguiere** [comment]
bot, build me.ahola.aphototoollibre

File: me.ahola.aphototoollibre.yml (hunk 2 above)
Comment (aphototool) [reply me.ahola.aphototoollibre.yml:9]: Copied from here https://docs.flatpak.org/en/latest/qt.html

Should line 8 be
 - --socket=fallback-x11
 
?

File: me.ahola.aphototoollibre.yml (hunk 3 above)
Comment (aphototool) [reply me.ahola.aphototoollibre.yml:21]: Yes, but that will require work on my .pro file that I like to make later.

File: me.ahola.aphototoollibre.yml (hunk 4 above)
Comment (aphototool) [reply me.ahola.aphototoollibre.yml:26]: Red lines worked in my tests. Do they fail in some situation?

Comment (hfiguiere) [reply me.ahola.aphototoollibre.yml:26]: it's coincidental they work.

File: me.ahola.aphototoollibre.yml (hunk 2 above)
Comment (hfiguiere) [reply me.ahola.aphototoollibre.yml:9]: this doc is wrong and doesn't ask the right questions.

File: me.ahola.aphototoollibre.yml (hunk 1 above)
Comment (hfiguiere) [reply me.ahola.aphototoollibre.yml:10]:
```suggestion
```

File: me.ahola.aphototoollibre.yml (hunk 3 above)
Comment (hfiguiere) [reply me.ahola.aphototoollibre.yml:20]: why so?

File: me.ahola.aphototoollibre.yml [hunk 5]
```diff
@@ -0,0 +1,26 @@
+app-id: me.ahola.aphototoollibre
+runtime: org.kde.Platform
+runtime-version: '5.15-22.08'
+sdk: org.kde.Sdk
+command: aphototoollibre
+finish-args:
+  - --share=ipc
+  - --socket=fallback-x11
+  - --socket=wayland
+  - --filesystem=host
```
Comment (hfiguiere) [inline me.ahola.aphototoollibre.yml:10]: filesystem=host is still a broad permission

**Comment by barthalion** [comment]
bot, build me.ahola.aphototoollibre

**Pull request closed without merging**